src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeExtensions.cs

```diff
@@ -65,45 +65,7 @@ public static Type GetReturnType(this MethodBase mi)
             return (mi.IsConstructor) ? mi.DeclaringType : ((MethodInfo)mi).ReturnType;
         }
 
-        public static TypeCode GetTypeCode(this Type type)
-        {
-            if (type == null)
-                return TypeCode.Empty;
-            else if (type == typeof(bool))
-                return TypeCode.Boolean;
-            else if (type == typeof(char))
-                return TypeCode.Char;
-            else if (type == typeof(sbyte))
-                return TypeCode.SByte;
-            else if (type == typeof(byte))
-                return TypeCode.Byte;
-            else if (type == typeof(short))
-                return TypeCode.Int16;
-            else if (type == typeof(ushort))
-                return TypeCode.UInt16;
-            else if (type == typeof(int))
-                return TypeCode.Int32;
-            else if (type == typeof(uint))
-                return TypeCode.UInt32;
-            else if (type == typeof(long))
-                return TypeCode.Int64;
-            else if (type == typeof(ulong))
-                return TypeCode.UInt64;
-            else if (type == typeof(float))
-                return TypeCode.Single;
-            else if (type == typeof(double))
-                return TypeCode.Double;
-            else if (type == typeof(decimal))
-                return TypeCode.Decimal;
-            else if (type == typeof(DateTime))
-                return TypeCode.DateTime;
-            else if (type == typeof(string))
-                return TypeCode.String;
-            else if (type.GetTypeInfo().IsEnum)
-                return Enum.GetUnderlyingType(type).GetTypeCode();
-            else
-                return TypeCode.Object;
-        }
+        public static TypeCode GetTypeCode(this Type type) => Type.GetTypeCode(type);
```

Comment:
We have many places where we perform a chain of type operations, a la `GetNonNullableType().GetTypeCode()`. The use of an extension method makes this fluent, and we don't have to scout around the code base to find all occurrences of `Type.GetTypeCode` to introduce/revert this dependency. It also should get inlined properly by the JIT. This said, after the dust has settled, we could totally go back to using `Type.GetTypeCode` if we expect never to have to worry about it disappearing again...